registry/client/repository.go

```diff
@@ -604,6 +621,45 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo string, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
As is this use of `string` is consistent with the rest of the client package. I would agree though that it would be desirable to refactor the interfaces to use the `reference` package, I would not enforce that here.